streams/src/main/java/org/apache/kafka/streams/processor/internals/SinkNode.java

```diff
@@ -82,7 +107,12 @@ public void process(final Record<KIn, VIn> record) {
 
         final String topic = topicExtractor.extract(key, value, contextForExtraction);
 
-        collector.send(topic, key, value, record.headers(), timestamp, keySerializer, valSerializer, partitioner);
+        final int bytesProduced =
+            collector.send(topic, key, value, record.headers(), timestamp, keySerializer, valSerializer, partitioner);
+
+        // Just use the cached system time to avoid the clock lookup
```

Comment:
I think it is fine as you do it since we do it similarly in other places like `processAtSourceSensor.record(1.0d, context.currentSystemTimeMs());` in  `SourceNode`.